lua/git-links/utils.lua

```diff
@@ -29,6 +33,23 @@ local function clean_url(url)
   return url
 end
 
+local function check_branch_remote()
+  local branch = vim.fn.system("git rev-parse --abbrev-ref HEAD"):gsub("%s+", "")
+  local remote_branches = vim.fn.system("git branch -r"):gsub("%s+", "")
+  if not remote_branches:find("origin/" .. branch, 1, true) then
+    error("Branch " .. branch .. " does not exist on the remote.", 0)
+  end
```

Comment:
It probably makes more sense to check whether the specific commit exists on the remote instead of the branch.  I think that you could do this with:

```
local remote_hash = vim.system({ "git", "branch", "--short", "HEAD", Utils.data.hash }, { cwd = vim.fn.expand("%:p:h") }):wait().stdout
if(remote_hash ~= "") then
  error("Current HEAD does not exist on the remote.", 0)
end
```

Function should likely be renamed to fit the new behavior as well.

Reply:
Thanks for taking a look! You're definitely right, and my initial approach is probably very inefficient for huge repos since it just lists all the branches. I wasn't able to get `git branch --short HEAD <hash>` to work, but I think it would probably be best to try and fetch a certain hash and then error if the sha isn't found. I updated the function and swapped the order to fetch the hash first. How does this logic look?

Reply:
Ah I'm an idiot.  The command itself to check the hash is:
`git branch -r --contains <hash>` so the code would be:
```
local remote_hash = vim.system({ "git", "branch", "-r", "--contains", Utils.data.hash }, { cwd = vim.fn.expand("%:p:h") }):wait().stdout
if(remote_hash ~= "") then
  error("Current HEAD does not exist on the remote.", 0)
end
```

Reply:
This could cause a false positive if there hasn't been a fetch lately, but I feel like it's unlikely that the local copy has seen a commit on the server but that it was somehow removed.  

Honestly not sure which way is better really.  I like the idea of doing a fetch on the hash since that actually checks with the server, though that does kind of introduce a network dependency for the plugin to work.  (i.e. non-functional if you're working offline).  Though I'm not sure if the plugin itself would error out or just provide an inaccurate message that the branch didn't exist on the remote.

Reply:
Unfortunately no internet connection and a failed fetch both result in the same exit code of `128`, so it would be tough to handle those cases differently. But how about just a try/catch like this? First try to fetch and then fall back to a `git branch -r` and if both fail raise an error. 

```lua
local function check_sha_remote()
  local success, result = pcall(function()
    return vim.system({ "git", "fetch", "origin", Utils.data.hash }, { cwd = vim.fn.expand("%:p:h") }):wait()
  end)

  if not success or result.code ~= 0 then
    local branch_result = vim.system({ "git", "branch", "-r", "--contains", Utils.data.hash },
      { cwd = vim.fn.expand("%:p:h") }):wait()
    if branch_result.stdout == "" then
      error("Current commit does not exist on remote branches.", 0)
    end
  end
end
```

Reply:
This works, but not with the shortened hash version.  I see two options to resolve this:
**Option 1: **
Switch to using the full hash 

**Option 2:**
Do a raw `git fetch --all` to fetch all remotes first and swallow any error it produces, then do the same local check for the short hash.

I keep going back and forth on which is best.  What do you think?

Reply:
Also, FYI, I really do appreciate this PR and I'm not trying to give you a hard time.  I'm very much learning as I go with this.  haha.

Reply:
I'm happy to contribute, thanks for the discussion! How about checking local first and then if that fails do a fetch, but I still think that fetching a specific sha is best because a full fetch against large repos could cause undesired side effects. some users may not want to fetch all just for this plugin's sake if they wouldn't fetch all otherwise.

Is there any downside to globally updating `Utils.data.hash` to be the full hash instead of the short? I just pushed another update with these changes for you to take a look at.

Reply:
Yeah I had the same concern about the fetch all.  On large repos that could be problematic.  I think your approach makes sense though I would reverse it.  Checking remote first then falling back to local if for some reason you can't check remote.  I think it's far more likely that a commit exists locally but hasn't been pushed than the reverse.

Reply:
Oh no I was wrong.  `git branch -r contains` does check if it exists on the remote base don when we last fetched.  So your PR makes total sense.  

Going to go ahead and approve it as I've got a PR adding tests that I've been waiting to merge because it will likely cause conflicts.